spec/gemstash/cli/authorize_spec.rb

```diff
@@ -148,4 +180,71 @@
       expect { Gemstash::CLI::Authorize.new(cli).run }.to raise_error(Gemstash::CLI::Error)
     end
   end
+
+  context "with the --list option" do
+    let(:cli_options) { { :list => true } }
+
+    it "lists un-named authorizations" do
+      Gemstash::Authorization.authorize("auth-key-all", "all")
+      Gemstash::Authorization.authorize("auth-key-push", %w[push])
+      Gemstash::Authorization.authorize("auth-key-yank", %w[yank])
+      Gemstash::Authorization.authorize("auth-key-fetch", %w[fetch])
+
+      Gemstash::CLI::Authorize.new(cli).run
+      expect(@said).to match(/auth-key-all.*all/)
+      expect(@said).to match(/auth-key-push.*push/)
+      expect(@said).to match(/auth-key-yank.*yank/)
+      expect(@said).to match(/auth-key-fetch.*fetch/)
+    end
+
+    it "lists named authorizations" do
+      Gemstash::Authorization.authorize("auth-key-all", "all", "auth all")
+      Gemstash::Authorization.authorize("auth-key-push", %w[push], "auth push")
+      Gemstash::Authorization.authorize("auth-key-yank", %w[yank], "auth yank")
+      Gemstash::Authorization.authorize("auth-key-fetch", %w[fetch], "auth fetch")
+
+      Gemstash::CLI::Authorize.new(cli).run
+      expect(@said).to match(/auth all.*auth-key-all.*all/)
+      expect(@said).to match(/auth push.*auth-key-push.*push/)
+      expect(@said).to match(/auth yank.*auth-key-yank.*yank/)
+      expect(@said).to match(/auth fetch.*auth-key-fetch.*fetch/)
+    end
+
+    it "lists specific authorization by name" do
+      Gemstash::Authorization.authorize("auth-key-all", "all", "auth all")
+      Gemstash::Authorization.authorize("auth-key-push", %w[push], "auth push")
+      cli_options[:name] = "auth push"
+
+      Gemstash::CLI::Authorize.new(cli).run
+      expect(@said).not_to match(/auth all.*auth-key-all.*all/)
+      expect(@said).to match(/auth push.*auth-key-push.*push/)
+    end
+
+    it "lists specific authorization by key" do
+      Gemstash::Authorization.authorize("auth-key-all", "all", "auth all")
+      Gemstash::Authorization.authorize("auth-key-push", %w[push], "auth push")
+      cli_options[:key] = "auth-key-push"
+
+      Gemstash::CLI::Authorize.new(cli).run
+      expect(@said).not_to match(/auth all.*auth-key-all.*all/)
+      expect(@said).to match(/auth push.*auth-key-push.*push/)
+    end
+
+    it "errors when auth is specified by both name and key" do
+      Gemstash::Authorization.authorize("auth-key-all", "all", "auth all")
+      Gemstash::Authorization.authorize("auth-key-push", %w[push], "auth push")
+      cli_options[:name] = "auth push"
+      cli_options[:key] = "auth-key-push"
+
+      expect { Gemstash::CLI::Authorize.new(cli).run }.to raise_error(Gemstash::CLI::Error)
+    end
+
+    it "handles missing authorization" do
```

Comment:
Perhaps this should express "raises an error on missing" so that we describe its behaviour.

Reply:
Sure thing, fixed.